app/scripts/modules/core/src/slack/SlackChannelSelector.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import * as Select from 'react-select';
+import VirtualizedSelect from 'react-virtualized-select';
+
+import { ISlackChannel, SlackReader } from './SlackReader';
+
+export interface ISlackChannelSelectorProps {
+  channel: ISlackChannel;
+  callback: (name: string, value: any) => void;
+}
+
+export interface ISlackChannelSelectorState {
+  channels: ISlackChannel[];
+  selected: ISlackChannel;
+  loading: boolean;
+}
+
+export default class SlackChannelSelector extends React.Component<
+  ISlackChannelSelectorProps,
+  ISlackChannelSelectorState
+> {
+  public constructor(props: ISlackChannelSelectorProps) {
+    super(props);
+    this.state = {
+      channels: [],
+      selected: this.props.channel || null,
+      loading: true,
+    };
+  }
+
+  public componentDidMount() {
+    SlackReader.getChannels().then((channels: ISlackChannel[]) =>
+      this.setState({
+        channels,
+        loading: false,
+      }),
+    );
+  }
+
+  private onChange = (evt: Select.Option<ISlackChannel>) => {
+    const channel = evt ? evt.value : null;
+    const { callback } = this.props;
+    callback('slackChannel', channel || {});
+    this.setState({ selected: channel });
+  };
+
+  public render() {
+    const { channels, loading, selected } = this.state;
+
+    return (
+      <div className="form-group row">
+        <div className="col-sm-3 sm-label-right">Slack Channel</div>
+        <div className="col-sm-9">
+          <VirtualizedSelect
```

Comment:
Can this be migrated to `<ReactSelectInput mode="VIRTUALIZED" />`?  I'd like to move towards using a more opinionated set of  components across the board.

Reply:
Yes, did not see this component so I will swap it out.